server/sonar-web/src/main/js/apps/projects/components/PageSidebar.tsx

```diff
@@ -39,26 +37,26 @@ import TagsFilter from '../filters/TagsFilter';
 import { translate } from '../../../helpers/l10n';
 import { RawQuery } from '../../../helpers/query';
 import { Facets } from '../types';
+import ClearAll from './ClearAll';
 
 interface Props {
   facets?: Facets;
-  isFavorite: boolean;
+  onClearAll: () => void;
+  onQueryChange: (change: RawQuery) => void;
   organization?: string;
   query: RawQuery;
   view: string;
   visualization: string;
 }
 
 export default function PageSidebar(props: Props) {
-  const { facets, query, isFavorite, organization, view, visualization } = props;
+  const { facets, onQueryChange, query, organization, view, visualization } = props;
   const isFiltered = Object.keys(query)
     .filter(key => !['view', 'visualization', 'sort'].includes(key))
     .some(key => query[key] != null);
   const isLeakView = view === 'leak';
-  const basePathName = organization ? `/organizations/${organization}/projects` : '/projects';
-  const pathname = basePathName + (isFavorite ? '/favorite' : '');
   const maxFacetValue = getMaxFacetValue(facets);
-  const facetProps = { isFavorite, maxFacetValue, organization, query };
+  const facetProps = { onQueryChange, maxFacetValue, organization, query };
 
   let linkQuery: RawQuery | undefined = undefined;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "linkQuery". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1854)